pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -343,6 +348,45 @@ type AlertmanagerWebSpec struct {
 	Timeout *uint32 `json:"timeout,omitempty"`
 }
 
+// GlobalSMTPConfig configures smtp parameters that are valid in all other configuration contexts.
+// See https://prometheus.io/docs/alerting/latest/configuration/#configuration-file
+type GlobalSMTPConfig struct {
+	// The default SMTP From header field.
+	// +optional
+	From *string `json:"from,omitempty"`
+
+	// The default SMTP smarthost used for sending emails, including port number.
+	// Port number usually is 25, or 587 for SMTP over TLS (sometimes referred to as STARTTLS).
+	// Example: smtp.example.org:587
+	// +optional
+	SmartHost *string `json:"smartHost,omitempty"`
```

Comment:
(suggestion) we can add some validations to the field
```
	SmartHost *HostPort `json:"smartHost,omitempty"`

...

// +kubebuilder:validation:Pattern=`^.+:.+$`
type HostPort string
```

The regexp validation isn't 100% proof but it should be good enough for catching obvious mistakes.

The other option is to store host & port in a struct:

```
	SmartHost *NetworkAddress `json:"smartHost,omitempty"`

...

type NetworkAddress struct {
    // +kubebuilder:validation:MinLength=1
    Host string `json:"host,omitempty"`
    // +kubebuilder:validation:MinLength=1
    Port string `json:"port,omitempty"`
}
```